### PR TITLE
Add implementation of Longest Common Subsequence (LCS) algorithm in Python

### LINKED ISSUE #117: [Feature Request] Adding Longest Common Subsequence algorithm
### Describe the solution you'd like
I want to add a new algorithm under DSA section and under dynamic programming. 

### Describe alternatives you've considered

**Would you like to work on this feature?**  
- [x] Yes  
- [ ] No  

I want to contribute under gssoc-ext and hacktoberfest.
Please assign this to me.

### PR DESCRIPTION
**Pull request by tanushrigoel**
### Description
Added documentation for longest common subsequence using dynamic programming

Fixes #117

### Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update

### Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Comment by tanushrigoel** [comment]
Please add hacktoberfest and gssoc-ext label

File: docs/dynamic-programming/longest_common_subsequence.md [hunk 1]
```diff
@@ -0,0 +1,121 @@
+
+# Longest Common Subsequence (LCS)
```
Comment (ajay-dhangar) [inline docs/dynamic-programming/longest_common_subsequence.md:9]: We need on top

```md
---
id: ___
title: ___
sidebar_label: ___
sidebar_position: ___
description: ___
tags: [___, __, ]
---
```

### Detailed Guidelines

- id: Unique identifier for the route.
- title: Title of the problem.
- sidebar_label: Problem number or label for the sidebar.
- tags: Relevant tags for the problem.
- description: A short SEO-friendly description. No HTML tags.

File: docs/dynamic-programming/longest_common_subsequence.md [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+## Dynamic Programming Approach
+
+To solve the LCS problem, we can utilize dynamic programming by creating a 2D table where `dp[i][j]` represents the length of the LCS of the first `i` characters of string X and the first `j` characters of string Y.
+
+### Recurrence Relation
+
+1. If the characters of X and Y match (i.e., `X[i-1] == Y[j-1]`), then:
+   \[
+   dp[i][j] = dp[i-1][j-1] + 1
+   \]
```
Comment (ajay-dhangar) [inline docs/dynamic-programming/longest_common_subsequence.md]: replace

```md
\[
   dp[i][j] = dp[i-1][j-1] + 1
   \]
```

with

```md
$$
 dp[i][j] = dp[i-1][j-1] + 1
$$
```

File: docs/dynamic-programming/longest_common_subsequence.md [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+2. If the characters do not match, we take the maximum value from either ignoring the current character from X or Y:
+   \[
+   dp[i][j] = \max(dp[i-1][j], dp[i][j-1])
+   \]
```
Comment (ajay-dhangar) [inline docs/dynamic-programming/longest_common_subsequence.md]: replace 

```md
   \[
   dp[i][j] = \max(dp[i-1][j], dp[i][j-1])
   \]
```

with

```md
  $$
   dp[i][j] = \max(dp[i-1][j], dp[i][j-1])
   $$
```

File: docs/dynamic-programming/longest_common_subsequence.md [hunk 4]
```diff
@@ -0,0 +1,121 @@
...
+    -   If the characters do not match, we set `dp[i][j]` to the maximum of the values from the cell above (`dp[i - 1][j]`) and the cell to the left (`dp[i][j - 1]`).
+3.  **Returning the Result:**
+    
+    -   Once the table is completely filled, the value at `dp[m][n]` contains the length of the longest common subsequence.
+
+
+
+## Time and Space Complexity
+
+-   **Time Complexity:** The time complexity is O(m * n) because we fill up a table of size `(m + 1) x (n + 1)` in nested loops.
```
Comment (ajay-dhangar) [inline docs/dynamic-programming/longest_common_subsequence.md]: replace `O(m * n)` with  `$O(m \times n)$`

File: docs/dynamic-programming/longest_common_subsequence.md [hunk 5]
```diff
@@ -0,0 +1,121 @@
...
+-   **Space Complexity:** The space complexity is also O(m * n) due to the 2D table used for storing intermediate results.
```
Comment (ajay-dhangar) [inline docs/dynamic-programming/longest_common_subsequence.md]: replace `O(m * n)` with `$O(m \times n)$`

**Review by ajay-dhangar — CHANGES_REQUESTED** [review]
update your code based on suggestions

File: docs/dynamic-programming/longest_common_subsequence.md (hunk 1 above)
Comment (tanushrigoel) [reply docs/dynamic-programming/longest_common_subsequence.md:9]: I am not sure what to add in id, sidebar_label. Can i provide them on my own? Or maybe see somewhere

**Review by ajay-dhangar — APPROVED** [review]
Done

**Pull request merged**